Extensions/xkit_patches.js

```diff
@@ -331,44 +389,73 @@ XKit.tools.getParameterByName = function(name){
 
 		XKit.iframe = {
 
+			/**
+			 * @return {String} Id of blog which the iframe refers to (usually
+			 *                  the blog in which the iframe is embedded)
+			 */
 			get_tumblelog: function() {
 				var new_channel_id = document.location.href.match(/[&?]tumblelogName=([\w-]+)/);
 				var old_channel_id = $("#tumblelog_name").attr('data-tumblelog-name');
 				return (new_channel_id && new_channel_id[1]) || old_channel_id;
 			},
 
+			/**
+			 * @return {String?} Post which this iframe refers to or undefined
```

Comment:
cannot return undefined—will throw a TypeError if not run in an iframe. (this is normally never an issue, as all users currently use other methods to determine if they're in a post or not, but i'm down for changing if you want to)